### PR TITLE
Add support for specifying a specific HTTP Host: header on the origin.

### LINKED ISSUE #111: Support setting http Host header
Support setting 'Host: target' http headers when creating tunnel.

A number of web targets rely on vhost/host matching to properly operate.

### PR DESCRIPTION
**Pull request by davebarrau**
Potentially resolves #111 

Very quick implementation, didn't test everything, but what I did test worked for me.

Usage: cloudflared tunnel -url http://localhost -httphost example.test

Can someone please check it?

### CODE REVIEW

**Comment by davidkarlsen** [comment]
@davebarrau do you have a binary to run=

File: cmd/cloudflared/tunnel/cmd.go [hunk 1]
```diff
@@ -488,6 +488,12 @@ func tunnelFlags(shouldHide bool) []cli.Flag {
 			EnvVars: []string{"TUNNEL_HOSTNAME"},
 			Hidden:  shouldHide,
 		}),
+		altsrc.NewStringFlag(&cli.StringFlag{
+			Name:    "httphost",
+			Usage:   "Sets the HTTP Host header for the local webserver.",
+			EnvVars: []string{"TUNNEL_HTTPHOST"},
+			Hidden:  shouldHide,
+		}),
```
[the remark this answers is not in the document]
Comment (davebarrau) [reply cmd/cloudflared/tunnel/cmd.go]: Done.

**Comment by davebarrau** [comment]
> Many thanks for submitting the PR! It looks great, I have left minor comments and will be happy to merge it once they're resolved.

Hi @sssilver, changes have been made. I also forgot to add the Host: header to serveWebsocket(), so that has been done too.

> Additionally, do you mind including the issue number in the commit message to make it easier to see context for commit in the future?

No worries, will remember this for future.

Thanks!

Dave

**Comment by davidkarlsen** [comment]
@sssilver Can this be merged now?

**Pull request merged**